x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/huggingface/elser/HuggingFaceElserService.java

```diff
@@ -41,9 +41,6 @@ public class HuggingFaceElserService implements InferenceService {
     private final SetOnce<HttpRequestSenderFactory> factory;
     private final SetOnce<ThrottlerManager> throttlerManager;
     private final AtomicReference<Sender> sender = new AtomicReference<>();
-    // This is initialized once which assumes that the settings will not change. To change the service, it
-    // should be deleted and then added again
-    private final AtomicReference<HuggingFaceElserAction> action = new AtomicReference<>();
```

Comment:
I don't think this is working like I was hoping it was. I believe this would set the action with the model for the first hugging face service created. Since it's not null after that it wouldn't work right if a new hugging face service was created.

I'm switching this to just create the action during the `infer` call. That might be wasteful because of short lived objects. If we want to improve that later we could create a concurrent hashmap to cache the actions and remove them when a service is deleted.